### PR TITLE
Improve CLI Help Text Descriptions

### PR DESCRIPTION
**Pull request by praneesha**
## Purpose
Improve CLI help text descriptions.

Fixes #<Issue Number>

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-test.help [hunk 1]
```diff
@@ -6,90 +6,92 @@ SYNOPSIS
...
 
 DESCRIPTION
-       Test compiles and executes Ballerina test sources and prints a summary of
-       test results. You can execute tests in a Ballerina package or a standalone
+       Compile and execute Ballerina test sources and print a summary of
+       the test results. You can execute tests in a Ballerina package or a standalone
        Ballerina file.
     
-       Use 'bal test to execute tests in the current package.
+       Use 'bal test` to execute the tests in the current package.
```
Comment (Dilhasha) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-test.help:13]:
```suggestion
       Use 'bal test' to execute the tests in the current package.
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-encrypt.help [hunk 2]
```diff
@@ -6,8 +6,8 @@ SYNOPSIS
 
 
 DESCRIPTION
-       Encrypt command performs symmetric encryption of sensitive
-       configuration values using AES, CBC mode with PKCS#5 padding.
+       Perform symmetric encryption of sensitive
```
Comment (ldclakmal) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-encrypt.help:9]:
```suggestion
       Perform AES symmetric encryption of sensitive
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-encrypt.help [hunk 3]
```diff
@@ -6,8 +6,8 @@ SYNOPSIS
 
 
 DESCRIPTION
```
Comment (ldclakmal) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-encrypt.help:8]: I don't think this command is there anymore. @warunalakshitha

Comment (warunalakshitha) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-encrypt.help:8]: Yes. We need to remove this. Seems missed once we removed the encrypt command.

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-build.help [hunk 4]
```diff
@@ -43,24 +40,23 @@ OPTIONS
...
            Generate an HTML report containing the test results. Defaults to 'true'
            if code coverage is enabled.
 
        --debug
-           Run tests in remote debugging mode, only when the '--skip-tests' flag is not used.
+           Run tests in remote debugging mode only when the '--skip-tests' flag is not used.
 
        --observability-included
-           Package observability in the executable JAR file(s).
+           Include package observability in the executable JAR file(s).
```
Comment (pubudu91) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-build.help:53]: This may need a bit more elaboration. Doesn't really add anything that the flag itself doesn't convey. What exactly does it mean to have observability enabled?

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-help.help [hunk 5]
```diff
@@ -8,31 +8,31 @@ SYNOPSIS
...
-       Ballerina is a statically typed, concurrent programming language, focusing on network interaction and structured
-       data. It is intended to be the core of a language-centric middleware platform. It has all the general-purpose
-       functionality expected of a modern programming language, but it also has several unusual aspects that make it
+       Ballerina is a statically-typed, concurrent programming language focusing on network interaction and structured
+       data. It is intended to be the core of a language-centric middleware platform. It not only has all the general-purpose
+       functionality expected from a modern programming language, but also has several unusual aspects that make it
        particularly suitable for its intended purpose.
 
-       Find more information at: https://ballerina.io/
+       Find more information at: https://ballerina.io/.
```
Comment (pubudu91) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-help.help:16]:
```suggestion
       Find more information at: https://ballerina.io/
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-test.help (hunk 1 above)
Comment (praneesha) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-test.help]: Backticks should be used here. Fixed.

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-help.help (hunk 5 above)
Comment (praneesha) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-help.help:16]: @pubudu91 - What is the reason to not have the slash and the full stop here?

**Review by Dilhasha — APPROVED** [review]

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-encrypt.help (hunk 3 above)
Comment (praneesha) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-encrypt.help:8]: Removed the `ballerina-encrypt.help` file.

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-encrypt.help (hunk 2 above)
Comment (praneesha) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-encrypt.help:9]: This is not valid since we are removing the `ballerina-encrypt.help` file.

**Review by warunalakshitha — DISMISSED** [review]
runtime related help test changes LGTM

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-help.help (hunk 5 above)
Comment (pubudu91) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-help.help:16]: Slash is not required. Plus a full stop there would mean it's part of the URL.

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-build.help (hunk 4 above)
Comment (pubudu91) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-build.help:53]: @nadundesilva

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-build.help [hunk 6]
```diff
@@ -43,24 +40,23 @@ OPTIONS
...
 
        --cloud
-           Enable cloud artifact generation for cloud providers such as kubernetes.
+           Enable cloud artifact generation for cloud providers such as Kubernetes.
```
Comment (anuruddhal) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-build.help:56]: shall we add supported values as well? 
`--cloud=docker` and `--cloud=k8s`

**Review by anuruddhal — CHANGES_REQUESTED** [review]

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-build.help [hunk 7]
```diff
@@ -6,33 +6,30 @@ SYNOPSIS
 
 
 DESCRIPTION
-       Build compiles Ballerina sources and produces executable JAR file(s). However, 
+       Compile Ballerina sources and produce executable JAR file(s). However,
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-build.help:9]: We only create one executable jar. So we can change the description as below:

```suggestion
       Compile Ballerina sources and produce the executable JAR file. However, 
```

**Review by lnash94 — APPROVED** [review]

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-build.help (hunk 6 above)
Comment (praneesha) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-build.help]: Updated.

**Review by anuruddhal — APPROVED** [review]

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-build.help (hunk 4 above)
Comment (nadundesilva) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-build.help:53]: There are a set of dependencies (Jars, Ballerina module imports & sytax tree json) which gets added to the final executable if this flag is provided. As of now this is the only difference that gets introduced when this flag is provided.

There is a **future plan** to add the compiler instrumentation only if the flag is provided as well (this will be a non breaking change as currently Observability cannot be used without this flag).

So what we expect from this flag is to package everything related to Observability so that it can be used (Reference doc on starting Prometheus - https://central.ballerina.io/ballerinax/prometheus). This flag was introduced as a requirement to reduce the Jar sizes as well as Code injections completely when Observability is not required.

**Review by warunalakshitha — APPROVED** [review]

**Review by ldclakmal — DISMISSED** [review]

**Review by azinneera — APPROVED** [review]
Package Commands changes LGTM

**Review by lnash94 — DISMISSED** [review]

**Review by Dilhasha — DISMISSED** [review]

**Review by daneshk — DISMISSED** [review]

**Review by hemikak — APPROVED** [review]

**Review by Dilhasha — APPROVED** [review]
Test command help changes LGTM.

**Review by parkavi11 — APPROVED** [review]

**Review by daneshk — APPROVED** [review]

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-build.help (hunk 4 above)
Comment (praneesha) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-build.help]: Updated the description.

**Review by parkavi11 — APPROVED** [review]

**Review by daneshk — APPROVED** [review]
gRPC CLI Help LGTM

**Review by lnash94 — DISMISSED** [review]

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help [hunk 8]
```diff
@@ -13,32 +13,34 @@ DESCRIPTION
        Generate Ballerina bridge code for Java APIs.
 
        Ballerina bindings could be generated for Java classes residing inside Java libraries
-       or for standard Java classes. The Java classes will be mapped to the Ballerina
-       classes providing a seamless Java interoperability developer experience to Ballerina users.
+       or standard Java classes. The Java classes will be mapped onto Ballerina
+       objects making the developer experience of Ballerina Java interoperability seamless.
```
Comment (pubudu91) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help:17]: Why change the phrasing here? "Ballerina objects" and "Ballerina classes" have 2 very different meanings. @IrushiL Please confirm the correct term to be used here.

Comment (praneesha) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help:17]: Seems like these changes were added when resolving the conflict caused by https://github.com/ballerina-platform/ballerina-lang/pull/31191/files. @IrushiL - Please confirm.

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help [hunk 9]
```diff
@@ -13,32 +13,34 @@ DESCRIPTION
...
 
-       In addition to the user-specified Java classes, partial implementations of directly
-       -dependent Java classes will also be generated by the tool. By default, the bindings for
-       each Java package will be mapped to a separate Ballerina module.
+       The directly-dependent Java classes and other required resources will be automatically
+       generated apart from the specified Java classes.
```
Comment (pubudu91) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help:20]: Why drop the remark about bindings of a Java package being mapped to a Ballerina module? Even the other one, conveys a completely different meaning to what the original text conveyed. We are not generating Java classes. We are generating Ballerina code for the Java classes.

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help [hunk 10]
```diff
@@ -13,32 +13,34 @@ DESCRIPTION
...
 OPTIONS
        (-cp|--classpath) <classpath>...
-           One or more comma-delimited classpaths for obtaining the jar files required for
+           One or more comma-delimited classpaths for obtaining the JAR files are required for
            generating the Ballerina bindings. This is not needed if the Ballerina bridge code
            is to be generated for standard Java classes.
 
        (-mvn|--maven) <groupId>:<artifactId>:<version>
-           Specify a Maven dependency with a colon-delimited group ID, artifact ID, and version.
+           A maven dependency with the colon-delimited groupId, artifactId, and version.
```
Comment (pubudu91) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help:30]: "Maven" is correct since it's a name of a tool. And the rest of the previous names/terms were correct as well.

File: cli/ballerina-cli/src/main/resources/cli-help/internal-error-message.txt [hunk 11]
```diff
@@ -1,9 +1,10 @@
 ballerina: Oh no, something really went wrong. Bad. Sad.
 
-There should be a file named "ballerina-internal.log" in the current directory.
-If you are able to share with us the code that broke Ballerina then
-we would REALLY appreciate if you would report this to us:
-go to https://github.com/ballerina-platform/ballerina-lang/issues and
-create a bug report with both this log file and the sample code.
+We appreciate it if you can report the code that broke Ballerina in 
+https://github.com/ballerina-platform/ballerina-lang/issues with the 
+`ballerina-internal.log` file in your current directory and your sample code.
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/resources/cli-help/internal-error-message.txt:5]: We don't create the `ballerina-internal.log` file now. Shall we reword it to say: `with this log and your sample code`

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-run.help [hunk 12]
```diff
@@ -7,65 +7,62 @@ SYNOPSIS
...
-       Run builds and executes the given `.bal` file, package, or program.
+       Build and execute the given `.bal` file, package, or program.
 
-       If a Ballerina source file (`.bal` file) or a package directory is given, the run
+       If a Ballerina source file (`.bal` file) or a package directory is given, this
        command builds and runs it. Compilation is done internally and the
        executable will not be created.
 
        You may use the 'bal build' command to compile a source and
        provide the generated binary file (.jar file) to the run command.
```
Comment (pubudu91) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-run.help:17]: `.jar`?

File: cli/ballerina-cli/src/main/resources/cli-help/internal-error-message.txt [hunk 13]
```diff
@@ -1,9 +1,10 @@
...
+
+We thank you for helping make us better.
+
+
```
Comment (pubudu91) [inline cli/ballerina-cli/src/main/resources/cli-help/internal-error-message.txt:9]: This has changed. @azinneera

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help (hunk 9 above)
Comment (keizer619) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help:20]: @IrushiL

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help (hunk 10 above)
Comment (keizer619) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help:30]: Yes it has to be `Maven`

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help (hunk 8 above)
Comment (IrushiL) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help:17]: It should be `Ballerina classes`.

In the conflicting PR, we have replaced the outdated term with the new one (some time ago we generated Ballerina objects now we generate Ballerina classes).

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help (hunk 9 above)
Comment (IrushiL) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help:20]: Yeah seems like the older version is restored after resolving the conflicting PR.

Shall we change that as,

**In addition to the user-specified Java classes, partially implemented bindings for dependent Java classes will also be generated by the tool. By default, the bindings for each Java package will be mapped to a separate Ballerina module.**

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help [hunk 14]
```diff
@@ -13,32 +13,34 @@ DESCRIPTION
...
-       -dependent Java classes will also be generated by the tool. By default, the bindings for
-       each Java package will be mapped to a separate Ballerina module.
+       The directly-dependent Java classes and other required resources will be automatically
+       generated apart from the specified Java classes.
 
 
 OPTIONS
        (-cp|--classpath) <classpath>...
-           One or more comma-delimited classpaths for obtaining the jar files required for
+           One or more comma-delimited classpaths for obtaining the JAR files are required for
```
Comment (IrushiL) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help:25]: This describes what the `classpath` option is. Therefore, we should remove the additional word "are" since it does not mean the `classpath` option is mandatory. It just means you can add it if it is required for generating the bindings.
```suggestion
           One or more comma-delimited classpaths for obtaining the JAR files required for
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help (hunk 10 above)
Comment (IrushiL) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help:30]: Let's leave it like this. This change also seems to be affected by the merge conflict.

**A Maven dependency with a colon-delimited group ID, artifact ID, and version.**

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help [hunk 15]
```diff
@@ -13,32 +13,34 @@ DESCRIPTION
...
-           instances where all the mappings are required inside a single module. The
-           specified directory doesn't need to be inside a Ballerina package.
+       (-o|--output) <output>
+           Location of the generated Ballerina bridge code. If this path is not specified,
+           the output will be written to the same directory from which the command is run.
+
+       (-m|--modules)
+           The flag that can be used to generate separate Ballerina module mappings for each
+           Java package inside a project, instead of generating all the bindings inside a
+           single output directory.
```
Comment (IrushiL) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help:40]: This should be removed, it is no longer applicable.

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help [hunk 16]
```diff
@@ -13,32 +13,34 @@ DESCRIPTION
...
+           A maven dependency with the colon-delimited groupId, artifactId, and version. 
            This would resolve the transitive dependencies and update the `Ballerina.toml` file.
 
-       (-o|--output) <output-path>
-           Generate all bindings inside a single directory. This option could be used in
-           instances where all the mappings are required inside a single module. The
-           specified directory doesn't need to be inside a Ballerina package.
+       (-o|--output) <output>
+           Location of the generated Ballerina bridge code. If this path is not specified,
+           the output will be written to the same directory from which the command is run.
```
Comment (IrushiL) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help:35]: Let's leave the wording in the conflicting PR here as well, the meaning was updated there.

**Generate all bindings inside a single directory. This option could be used in instances where all the mappings are required inside a single module. The specified directory doesn't need to be inside a Ballerina package.**

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help [hunk 17]
```diff
@@ -13,32 +13,34 @@ DESCRIPTION
...
+
+       (-m|--modules)
+           The flag that can be used to generate separate Ballerina module mappings for each
+           Java package inside a project, instead of generating all the bindings inside a
+           single output directory.
 
        --public
-           Set the visibility modifier of the generated binding objects to public. This flag
-           will be applicable only if the bindings are generated inside a single directory.
+           Set the visibility modifier of the generated binding objects to public.
```
Comment (IrushiL) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help:43]: This too needs to be preserved. 

**Set the visibility modifier of the generated binding objects to public. This flag will be applicable only if the bindings are generated inside a single directory.**

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help (hunk 8 above)
Comment (praneesha) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help:17]: Updated as "Ballerina classes".

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help (hunk 9 above)
Comment (praneesha) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help:20]: Updated.

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help (hunk 10 above)
Comment (praneesha) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help:30]: Fixed.

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-run.help (hunk 12 above)
Comment (praneesha) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-run.help:17]: Fixed.

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help (hunk 15 above)
Comment (praneesha) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help:40]: Removed.

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help (hunk 16 above)
Comment (praneesha) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help:35]: Fixed.

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help (hunk 17 above)
Comment (praneesha) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-bindgen.help:43]: Fixed.

File: cli/ballerina-cli/src/main/resources/cli-help/internal-error-message.txt (hunk 11 above)
Comment (praneesha) [reply cli/ballerina-cli/src/main/resources/cli-help/internal-error-message.txt:5]: Updated.

File: cli/ballerina-cli/src/main/resources/cli-help/internal-error-message.txt (hunk 13 above)
Comment (praneesha) [reply cli/ballerina-cli/src/main/resources/cli-help/internal-error-message.txt:9]: Fixed.

**Comment by praneesha** [comment]
> If this needs to be sent to beta2, needs to be rebased to `stage-slbeta2`

Will send a separate PR to the `stage-slbeta2` branch.

**Review by azinneera — APPROVED** [review]
Package Command changes LGTM
